udev/11-stratisd.rules

```diff
@@ -0,0 +1 @@
+ACTION=="add|change", ENV{DM_NAME}=="stratis-1-[0-9a-f]*-thin-fs-[0-9a-f]*", IMPORT{program}="stratis_uuids_to_names $env{DM_NAME}", SYMLINK+="$env{STRATIS_SYMLINK}", ENV{SYSTEMD_WANTS}+="stratisd.service"
```

Comment:
`SYSTEMD_WANTS` doesn't do what I think you think it does. :-) It does not influence at all when the udev rule runs.

But I don't think you need to order the rule: The `ENV{DM_NAME}=="stratis-1-[0-9a-f]*-thin-fs-[0-9a-f]*"` will only match for devices that have been created by stratisd, so the reminder of the rule will only execute when stratisd is running (or has already been running at least once if it ever exits after a idle period, which I think it doesn't, right?).
The remaining thing then is to make sure that `stratis_uuids_to_names` can actually communicate with stratisd, maybe via D-Bus, maybe via something else.

Reply:
I'm reading the documentation in more depth, and thinking over some possible solutions. My initial reaction is that providing a systemd unit file for our D-Bus service name of type `dbus` may help here. I'm still doing research about what action that unit file should trigger, but it may work to just trigger a synthetic change event for all of the Stratis filesystem devices once the D-Bus service comes up fully. Do you have any initial reactions to this?

Reply:
This may present some problems as the `dbus` type imposes a requirement on `dbus.socket` which you mentioned causes a dependency cycle below.

Reply:
I'm going to test if I can create a unit file simply for the D-Bus service separate from the stratisd service file that depends on `stratisd.service` and see how systemd reacts to that.

Reply:
>  I'm still doing research about what action that unit file should trigger, but it may work to just trigger a synthetic change event for all of the Stratis filesystem devices once the D-Bus service comes up fully. Do you have any initial reactions to this?

The goal is to allow entries in /etc/fstab for Stratis filesystems with the "auto" option, right?  That's why stratisd.service is `Before=local-fs-pre.target´. If the symlinks haven't been created by that time, people can't use them in /etc/fstab. That would be surprising, I think.

Reply:
@mvollmer We have addressed that problem without stratisd. I've created a minimal Rust CLI using the daemon code to be used in contexts that don't have D-Bus available like initramfs.